src/ObjCRuntime/Runtime.CoreCLR.cs

```diff
@@ -985,7 +985,7 @@ bool IEqualityComparer<IntPtr>.Equals (IntPtr x, IntPtr y)
 			int IEqualityComparer<IntPtr>.GetHashCode (IntPtr obj)
 			{
 				unchecked {
-					return (int) Hash (obj);
+					return HashCode.Combine(obj);
```

Comment:
Ah, this is a very special case too :)  It's a hash table where the user can specify its own hash function:

https://github.com/xamarin/xamarin-macios/blob/25efb739fc40a5c769765eb3af1cae70824b01d7/src/ObjCRuntime/Runtime.CoreCLR.cs#L955-L966

this means that we still need to call the provided hash function.